docs/agent/config.md

```diff
@@ -87,7 +87,191 @@ differently from Agent version 5.
 | `gce_updated_hostname` | v6 behaves like v5 with `gce_updated_hostname` set to true. May affect reported hostname, see [doc][gce-hostname] |
 | `collect_security_groups` | feature still available with the aws integration  |
 
+## [BETA] Encrypted secrets
 
+**This feature is in beta and its options or behaviour might break between
+minor or bugfix releases of the Agent.**
+
+This feature is only available on Linux at the moment.
+
+Starting with the agent 6.3.0 the agent is able to collect encrypted secrets in
+the configurations files from an external source. This allows users to use the
+agent alongside a secret management services (like `Vault` or other).
+
+This section will cover how to set up this feature.
+
+### Encrypting password in check configuration
+
+To fetch a secret from a check configuration simply use the `ENC[]` notation.
+This handle can be used as a *value* of any YAML fields in your configuration
+(not the key) in any section (`init_config`, `instances`, `logs`, ...).
+
+Encrypted secrets are supported in every configuration backend: file, etcd,
+consul ...
+
+Example:
+
+```yaml
+instances:
+  - server: db_prod
+    user: ENC[db_prod_user]
+    password: ENC[db_prod_password]
+```
+
+In the above example we have two encrypted  secrets : `db_prod_user` and
+`db_prod_password`. Those are the secrets **handle** and must uniquely identify
+a secret within your secrets management tool.
+
+Between the brackets every character is allowed as long as the configuration
+YAML is valid. This means you could use any format you want.
+
+Example:
+
+```
+ENC[{"env": "prod", "check": "postgres", "id": "user_password", "az": "us-east-1a"}]
+```
+
+In this example the secret handle is the string `{"env": "prod", "check":
+"postgres", "id": "user_password", "az": "us-east-1a"}`.
+
+**Autodiscovery**:
+
+Secrets are resolved **after** Autodiscovery template variables. This means you
+can use them in a secret handle to fetch secrets specific to a container.
+
+Example:
+
+```
+instances:
+  - server: %%host%%
+    user: ENC[db_prod_user_%%host%%]
+    password: ENC[db_prod_password_%%host%%]
+```
+
+### Fetching secrets
+
+To fetch the secrets from you configurations you have to provide an executable
+capable of fetching passwords from your secrets management tool.
+
+An external executable solved multiple issues:
+
+- Guaranties that the agent will never have more access/information than what
+  you transfer to it.
+- Maximum freedom and flexibility: allowing users to use any secrets management
+  tool (including closed sources ones) without having to rebuild the agent.
+- Solving the **initial trust** is a very complex problem as it usually change
+  from users to users. Each setup requires more or less control and might rely on
+  private tools. This shift the problem where it's the easier to solve, without
```

Comment:
* ~This shift the problem where it's the easier to solve~
* This shifts the problem to where it is easier to solve